### PR TITLE
Disable Cross-Namespace by default for IngressRoute provider

### PR DESCRIPTION
**Pull request by dtomcej**
### What does this PR do?

Disables the `allowCrossNamespace` by default for IngressRoute provider

### Motivation

Security

### More

- [x] Added/updated tests
- [x] Added/updated documentation

### CODE REVIEW

File: pkg/provider/kubernetes/crd/kubernetes_test.go [hunk 1]
```diff
@@ -28,10 +28,11 @@ func Bool(v bool) *bool { return &v }
...
 func TestLoadIngressRouteTCPs(t *testing.T) {
 	testCases := []struct {
-		desc         string
-		ingressClass string
-		paths        []string
-		expected     *dynamic.Configuration
+		desc                  string
+		ingressClass          string
+		paths                 []string
+		enabledCrossNamespace bool
```
Comment (dtomcej) [inline pkg/provider/kubernetes/crd/kubernetes_test.go:34]: remove

**Review by jbdoumenjou — APPROVED** [review]
LGTM

**Review by ldez — APPROVED** [review]
LGTM

**Review by rtribotte — APPROVED** [review]
LGTM

**Review by jbdoumenjou — APPROVED** [review]
LGTM

**Pull request merged**